### PR TITLE
Add Health Checks for Consul ECS

### PR DESCRIPTION
**Pull request by paragbhingre**
This PR adds the Consul ECS health checks to the application container. Customers can either add Consul Native Health Checks by providing value to `consulChecks` or ECS Health Checks by providing values to `healthCheck` parameter in ECSConsulMeshExtension.

### CODE REVIEW

File: lib/consul-mesh-extension.ts [hunk 1]
```diff
@@ -183,6 +202,9 @@ export class ECSConsulMeshExtension extends ServiceExtension {
         this.gossipEncryptKey = props.gossipEncryptKey;
         this.serviceDiscoveryName = props.serviceDiscoveryName;
         this.consulDatacenter = props.consulDatacenter || "dc1";
+        this.healthCheck = props.healthCheck;
+        this.consulChecks = props.consulChecks || undefined;
```
Comment (upparekh) [inline lib/consul-mesh-extension.ts:206]: I'm curious, can the user provide both `healthCheck` and `consulChecks` as well? If not, we should probably catch that!

Comment (paragbhingre) [reply lib/consul-mesh-extension.ts:206]: Actually they can provide both, just that consul has logic to accept only one at a time. if consul health checks are defined then they ignore ECS health checks. But that's a good question @upparekh . I will try to get an answer for when both the checks are defined what should happen?

File: lib/consul-mesh-extension.ts [hunk 2]
```diff
@@ -168,6 +183,10 @@ export class ECSConsulMeshExtension extends ServiceExtension {
     private gossipEncryptKey?: secretsmanager.ISecret;
     private serviceDiscoveryName: string;
     private consulDatacenter?: string;
+    private healthCheck?: ecs.HealthCheck;
+    private consulChecks?: any[];
+    private defaultedCheckContainers?: string;
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply lib/consul-mesh-extension.ts]: Done

File: lib/consul-mesh-extension.ts [hunk 3]
```diff
@@ -537,11 +590,17 @@ export class ECSConsulMeshExtension extends ServiceExtension {
     }
 }
 
-
 export class ConsulMeshsMutatingHook extends ContainerMutatingHook {
+
+    private healthCheck?: ecs.HealthCheck;
+
+    constructor(healthCheck?: ecs.HealthCheck) {
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply lib/consul-mesh-extension.ts]: Sure, made changes

File: lib/consul-mesh-extension.ts (hunk 1 above)
Comment (paragbhingre) [reply lib/consul-mesh-extension.ts:221]: So as per discussion with HC we decided to throw an error when both(consul and ECS) the checks are mentioned.

File: lib/consul-mesh-extension.ts [hunk 4]
```diff
@@ -291,6 +332,12 @@ export class ECSConsulMeshExtension extends ServiceExtension {
             }
         );
 
+        if(this.healthCheck && this.consulChecks){
+            throw new Error('Can not define both Consul Native Checks and ECS Health Checks');
```
Comment (iamhopaul123) [inline lib/consul-mesh-extension.ts:336]: nit
```suggestion
            throw new Error('Cannot define both Consul Native Checks and ECS Health Checks');
```

File: lib/consul-mesh-extension.ts [hunk 5]
```diff
@@ -27,6 +27,13 @@ export enum CloudProviders {
     HASHICORP_CLOUDPROVIDER = 'hcp'
 }
 
+export interface ContainerMutatingProps {
+    /**
+     * Application Container Health Checks
+     */
```
Comment (upparekh) [inline lib/consul-mesh-extension.ts:33]: nit: @default doc for this prop (basically all optional properties of an interface should have one)

File: lib/consul-mesh-extension.ts [hunk 6]
```diff
@@ -134,8 +147,25 @@ export interface ECSConsulMeshProps {
...
+
+     /**
+      * consul native checks
+      */
+     consulChecks?: any[];
+
+    /**
+     * ACL secret arn
+     */
+     aclSecretArn?: string;
```
Comment (upparekh) [inline lib/consul-mesh-extension.ts:168]: nit: Can we add @default docs to these as well!

**Review by upparekh — COMMENTED** [review]
Some very minor updates 🙏

File: lib/consul-mesh-extension.ts [hunk 7]
```diff
@@ -291,6 +342,12 @@ export class ECSConsulMeshExtension extends ServiceExtension {
             }
         );
 
+        if(this.healthCheck && this.consulChecks){
+            throw new Error('Cannot define both Consul Native Checks and ECS Health Checks');
+        }
```
Comment (upparekh) [inline lib/consul-mesh-extension.ts:347]: We should also add a unit test for verifying this behavior.

**Review by upparekh — COMMENTED** [review]
LGTM! Just a small suggestion!

**Review by upparekh — DISMISSED** [review]

**Comment by troy-ameigh** [comment]
Lgtm

**Review by troy-ameigh — APPROVED** [review]

**Pull request merged**